tests/app/questionnaire/test_completeness.py

```diff
@@ -0,0 +1,483 @@
+from tests.app.app_context_test_case import AppContextTestCase
+
+from app.data_model.answer_store import AnswerStore, Answer
+from app.questionnaire.location import Location
+from app.questionnaire.completeness import Completeness
+from app.questionnaire.questionnaire_schema import QuestionnaireSchema
+from app.utilities.schema import load_schema_from_params
+
+
+class TestCompleteness(AppContextTestCase): # pylint: disable=too-many-public-methods
+    def test_no_groups(self):
+        schema = load_schema_from_params('test', 'navigation_completeness')
+
+        schema.json['sections'][0]['groups'] = []
+
+        progress = Completeness(schema, AnswerStore(), [], [], metadata={})
+
+        progress_value = progress.get_state_for_section('coffee-section')
+        self.assertEqual(Completeness.NOT_STARTED, progress_value)
+
+    def test_not_started_section(self):
+        schema = load_schema_from_params('test', 'navigation_completeness')
+
+        routing_path = [
+            Location('coffee-group', 0, 'coffee'),
+        ]
+
+        progress = Completeness(schema, AnswerStore(), [], routing_path, metadata={})
+
+        progress_value = progress.get_state_for_section('coffee-section')
+        self.assertEqual(Completeness.NOT_STARTED, progress_value)
+
+    def test_started_section(self):
+        schema = load_schema_from_params('test', 'navigation_completeness')
+
+        completed_blocks = [
+            Location('coffee-group', 0, 'coffee'),
+        ]
+
+        routing_path = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+        ]
+
+        progress = Completeness(
+            schema, AnswerStore(), completed_blocks, routing_path, metadata={})
+
+        progress_value = progress.get_state_for_section('coffee-section')
+        self.assertEqual(Completeness.STARTED, progress_value)
+
+    def test_completed_section(self):
+        schema = load_schema_from_params('test', 'navigation_completeness')
+
+        completed_blocks = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+        ]
+
+        routing_path = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+        ]
+        progress = Completeness(
+            schema, AnswerStore(), completed_blocks, routing_path, metadata={})
+
+        progress_value = progress.get_state_for_section('coffee-section')
+        self.assertEqual(Completeness.COMPLETED, progress_value)
+
+    def test_all_sections_complete_none(self):
```

Comment:
The test name doesn't make sense on its own, until you read the subsequent 2 tests